aea/skills/base/core.py

```diff
@@ -422,26 +531,28 @@ def _add_protocol(self, directory: str, protocol_name: str):
 class HandlerRegistry(Registry):
     """This class implements the handlers registry."""
 
-    def __init__(self, context: Context) -> None:
+    def __init__(self) -> None:
         """
         Instantiate the registry.
 
         :return: None
         """
-        self._handlers = {}  # type: Dict[SkillId, Handler]
-        self.context = context
+        self._handlers = {}  # type: Dict[ProtocolId, Dict[SkillId, Handler]]
```

Comment:
Note, the handler needs to assign based on protocol_id not skill_id. However, for now we keep track of the skill_id.